Add Livebox router

cyr-ius opened the pull request:
## Description:

Add Orange Livebox (router)
Sensors for bandwith ans status connection
Device Trackers

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#10260

## Example entry for `configuration.yaml` (if applicable):
```yaml:
      livebox:
         host:
         port:
         username:
         password:
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.
  - [X] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

cyr-ius committed:
The livebox integration allows you to observe and control Orange Livebox router.

PaulAnnekov committed:
bump tuyaha 0.0.4 (#26303)

balloob committed:
Fix Alexa Report State (#26305)

* Fix Alexa Report State

* Forgot to save a file

cyr-ius committed:
Add Option Flow Handler for disable device tracker

cyr-ius committed:
Lint syntax

Kane610 committed:
deCONZ - Dont update entry if data is equal

cyr-ius committed:
Lint syntax (play again)

balloob committed:
Fix alexa bad temp sensors (#26307)

cyr-ius committed:
Lint syntax (again)

cyr-ius committed:
Add config_flows and requirements

tyjtyj committed:
Fix google_maps scan interval (#26328)

Reported on 
https://github.com/home-assistant/home-assistant/issues/26275

cyr-ius committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into livebox

cyr-ius committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into livebox

cyr-ius committed:
Livebox component

cyr-ius committed:
Execute script.hassfest

cyr-ius committed:
Merge branch 'component_livebox' into livebox

Cyr-ius commented:
I tried everything and I do not understand why I do not pass the config_flows check test
Can someone help me?

cyr-ius committed:
Update config_flows.py

Cyr-ius commented:
I found.
there is a problem between the validation step and the format check step in CI

Black add a comma end of array when i do a pre-commit in git while the validation step does not support it
I open an issue

Cyr-ius commented:
Open Issue : [26334](https://github.com/home-assistant/home-assistant/issues/26334)

fmartens committed:
Inverted rflink cover (#26038)

* Added InvertedRflinkCover class to support COCO/KAKU ASUN-650 devices

* Rename TYPE_NORMAL to TYPE_STANDARD

* Cleaning up code and removed unused imports

* Added unit tests for InvertedRflinkCover

* less if/else statements

* Autoresolve type for newkaku

* Updated tests for InvertedRflinkCover

* Added unit test for standard cover without specifying type

* Updated comments in unit tests

* Updated unit test configuration and comments to be more explanatory

* Restore variable names in first part of the unit test that have been changed during a search and replace

* Reformated the code according to 4de97ab

* remove blank lines at end of rflink test_cover.py

* Replace single with double quote in test_cover.py

* Replaced single quotes with double qoutes and fixed formatting

* Black improvements

* Reformated the code of the unit test.

* entity_type_for_device_id should return 'TYPE_STANDARD' instead of 'None'

Kane610 committed:
UniFi - Simplify getting controller from config entry (#26335)

* Simplify getting controller from config entry

* Lint ignore no longer needed

* Fix tests

alemuro committed:
Add BeeWi SmartClim BLE sensors (#26174)

* Add BeeWi SmartClim BLE temperature and humidity sensor

* Update missing CODEOWNERS and .coveragerc files

* Updated requirements file

* Update documentation

* Fixed requested changes and decoupled IO library

* Add unique_id property

* Improve unique_id

gadgetchnnel committed:
New template sensor attributes (#26127)

* updated sensor and test files

* Formatting fixes

* Updated attribute template code

* Black formatting

* Code improvements based on feedback on binary_sensor pull request

* Updated tests

* Remove duplicated code and fix tests

* Black formatting on tests

* Remove link from docstring

* Moved default to schema

* Formatting fix and change to use dict[key] to retrieve attribute_templates

emontnemery committed:
Add improvements of device_automation from frontend PR 3514 (#26295)

* Add improvements from frontend PR 3514

* Fix test

* Tweak

balloob committed:
Expose current direction properly on state machine (#26298)

* Expose current direction properly on state machine

* Fix template fan

balloob committed:
Expose current direction properly on state machine (#26298)

* Expose current direction properly on state machine

* Fix template fan

ChristianKuehnel committed:
added missing bluepy dependency for miflora (#26297)

fixes dependency issue #19362 (only part of the solution)

balloob committed:
Update translations

balloob committed:
Updated frontend to 20190901.0

balloob committed:
Bumped version to 0.98.2

balloob committed:
Merge pull request #26345 from home-assistant/rc

0.98.2

exxamalte committed:
USGS Earthquakes icon for geolocation entities (#26353)

* define icon

* updated tests

* added codeowner

* updated codeowners

scop committed:
Test with 3.6.1 in Travis (#26347)

https://github.com/home-assistant/architecture/issues/278

Commit:
Push to version 0.7.10 of denonavr (#26362)

JeffLIrion committed:
Bump androidtv to 0.0.26 and update tests (#26340)

* Move the patchers to a separate file

* Got a pytest test working (mostly)

* Checkpoint

* Switch to pytest for all tests

* Bump androidtv to 0.0.26 and update tests

* More robust patching

* Remove unused constants

* Combine two lines

* Add 2 additional checks

* Check that state objects are not None; add more description to tests

* Use f strings

costastf committed:
Update google_maps dependency and improve error message (#26361)

* updated dependency and made error message a bit more accurate.

* updated dependency and made error message a bit more accurate.

doudz committed:
Update zigpy_zigate to 0.2.0 (#26327)

* Update zigpy_zigate to 0.2.0

* Update requirements_all.txt

Kane610 committed:
String has nothing to do with class method naming (#26368)

Kane610 committed:
String has nothing to do with class method naming (#26368)

pvizeli committed:
Add token support

balloob committed:
Allow passing dictionaries to async_register_entity_service (#26370)

fabaff committed:
Upgrade pyhaversion to 3.1.0 (#26232)

fabaff committed:
Upgrade pyhaversion to 3.1.0 (#26232)

crhan committed:
Support new climate arch for zhong_hong (#26309)

* support new climate arch for zhong_hong

* use black to format

* mapping the states between lib and HA

* Add zhong_hong mapping constant

pvizeli committed:
Update azure-pipelines-wheels.yml

frenck committed:
Correct file permissions, removing executable bits (#26376)

amelchio committed:
Fix race during initial Sonos group construction (#26371)

* Fix race during initial Sonos group construction

* Update homeassistant/components/sonos/media_player.py

amelchio committed:
Fix race during initial Sonos group construction (#26371)

* Fix race during initial Sonos group construction

* Update homeassistant/components/sonos/media_player.py

frenck committed:
Use literal string interpolation in integrations A (f-strings) (#26377)

* Use literal string interpolation in integrations A (f-strings)

* Black

ThaStealth committed:
Remove solaredge_local duplicate code (#25941)

* Removed duplicate code

* Update sensor.py

Splitted exceptions into two seperate ones

* Update sensor.py

* Update sensor.py

* Update sensor.py

Fixed linting errors

* Update sensor.py

frenck committed:
Use literal string interpolation in integrations B-D (f-strings) (#26378)

frenck committed:
Use literal string interpolation in integrations E-G (f-strings) (#26379)

exxamalte committed:
GeoNet NZ Quakes Sensor (#26078)

* working version of status sensor

* changed unit of measurement

* align naming with feed source

* simplified sensor name

* fix potential issue during initialisation

* fixed tests

* changed icon to constant

* added tests for new sensor

* split tests for geolocation vs sensor

* fixed lint

* fixed pylint

* fixed test

* removed config entry id from attributes

* moved entity manager to component

* fix issue with multiple config entries overriding each other's data

* creating async tasks instead of awaiting each unloading

* moved manager to component

* correctly triggering update only when this component is loaded

* fixed tests after major code refactorings

* fixed pylint

* moved actual creation of new events to geolocation platform

* changed all timestamps to utc

* changed the way platforms are setup and manager is updated

* simplify assert statement

* changed the way waiting for unloading platforms

frenck committed:
Use literal string interpolation in integrations H-J (f-strings) (#26380)

frenck committed:
Removes executable but from hassfest codeowners (#26381)

pvizeli committed:
Update translations_upload

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

ualex73 committed:
Add support Slide cover (#25913)

* Add support GoSlide cover

* Fixed Parameters differ from overridden
Fixed Removed other pylint warnings

* Renamed GoSlide to Slide, because of Innovation in Motion rebranding

* Fixed codeowners file

* Fixed requirements file

* Removed pylint: disable=unused-argument
Removed DOMAIN not exist check
Changed if to min/max
Changed 3rd party import to top of the module
Removed timeout/retry parameters
Removed unused constants
Added check for discovery_info is none
Changed pass slide object instead of full hass object
Changed pass api object instead of full hass object
Added unique_id functionality
Removed entity_id/name properties
Removed supported_features/state functions

* Fixed unused variables

* Changed Slide API uses snake names
Changed Improved exception handling
Changed Updated Slide API to 0.50.0

* Changed moved exceptions into goslide-api
Changed retry setup into coroutine

* Changed str(err) to err
Changed invert if result to if not result

frenck committed:
Use literal string interpolation in integrations N-Q (f-strings) (#26391)

pvizeli committed:
Update OpenCV 4.1.1 / Numpy 1.17.1 (#26387)

frenck committed:
Use literal string interpolation in integrations T-W (f-strings) (#26394)

frenck committed:
Use literal string interpolation in integrations K-M (f-strings) (#26389)

frenck committed:
Use literal string interpolation in integrations R-S (f-strings) (#26392)

frenck committed:
Use literal string interpolation in integrations X-Z (f-strings) (#26395)

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

frenck committed:
Fixes invalid JSON files and whitespace corrections in YAML files (#26396)

johnluetke committed:
Refactor pihole integration (#25837)

* Adds tests for pi_hole integration

* Refactor pi_hole component to an integration supporting multiple platforms

* Adds mock of Hole dependency

* Aborts platform setup when discovery_info is none

* Removes use of monitored_conditions

* Adds integration setup test

* Removes PlatformNotReady check

* Adds sensor test

* Code review updates

* Refactor tests to assert state through hass

* Reorder imports

vaidyasr committed:
Change xiaomi camera conf_host to template (#25799)

* changed conf_host to template

Changed conf_host to template to accommodate dynamic ips

* Update camera.py

* Updated per comment

* update for black formatter

* black changes

croghostrider committed:
Fix Emulated Hue AttributeError: 'NoneType' object has no attribute '… (#26018)

* Fix Emulated Hue AttributeError: 'NoneType' object has no attribute 'lower'

* Fix debug

* Update error message

balloob committed:
Allow passing None as input_text config (#26409)

balloob committed:
Fix state report (#26406)

* Fix state report

* Update test

balloob committed:
Fix state report (#26406)

* Fix state report

* Update test

exxamalte committed:
IGN Sismologia icon for geolocation entities (#26408)

* define icon

* fixed tests

ehendrix23 committed:
Update to 0.1.13 (#26402)

Update to 0.1.13

ehendrix23 committed:
Update to 0.1.13 (#26402)

Update to 0.1.13

frenck committed:
Migrate legacy typehints in core to PEP-526 (#26403)

* Migrate legacy typehints in core to PEP-526

* Fix one type

taylorsilva committed:
Add X10 devices as ISY994 switches (#26342)

* Add X10 devices as switches

Signed-off-by: Taylor Silva <taylorsilva@outlook.com>

* Add logging message for when unsupported device is found.

This will help a user figure out why they aren't seeing all devices.

Signed-off-by: Taylor Silva <taylorsilva@outlook.com>

* Update homeassistant/components/isy994/__init__.py

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Update homeassistant/components/isy994/__init__.py

100 is too generic of an ISY uom. We don't want to accidentally add non-switch devices.

Quentame committed:
Add config flow to linky (#26076)

* Linky: setup ConfigFlow

* async_track_time_interval

* Review from @MartinHjelmare 1

* Review from @MartinHjelmare 2

* Review from @MartinHjelmare 3

* Review from @MartinHjelmare 4

* black --fast homeassistant tests

* Bump pylinky to 0.4.0 and add error user feedback

* Fix .coveragerc

* Linky platform moved to integration in config.yml and with multiple accounts

* Remove useless logs

* Review from @MartinHjelmare 5

* Add config flow tests

* Add config flow tests : login + fetch on failed

OverloadUT committed:
Bump ISY994's PyISY dependency to 1.1.2 (#26413)

Fixed a major bug that was responsible for ISY events getting seemingly random delays up to 24 seconds

OverloadUT committed:
Bump ISY994's PyISY dependency to 1.1.2 (#26413)

Fixed a major bug that was responsible for ISY events getting seemingly random delays up to 24 seconds

Danielhiversen committed:
Met, check for existing location (#26400)

Danielhiversen committed:
Met, check for existing location (#26400)

balloob committed:
Allow core config updated (#26398)

balloob committed:
Allow core config updated (#26398)

frenck committed:
Correct file permissions in slide integration (#26390)

zxdavb committed:
Use literal string interpolation in honeywell (#26386)

* Initial commit

* fix lint hints

zxdavb committed:
Initial commit (#26385)

zxdavb committed:
Initial commit (#26383)

balloob committed:
Bumped version to 0.98.3

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

pvizeli committed:
Merge pull request #26414 from home-assistant/rc

0.98.3

Commit:
[ci skip] Translation update

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

exxamalte committed:
NSW Rural Fire Service icon for geolocation entities (#26416)

* define icon

* add myself as codeowner

pvizeli committed:
Add prettier to vscode (#26417)

bramkragten committed:
Updated frontend to 20190904.0 (#26421)

* Updated frontend to 20190904.0

* Updated frontend to 20190904.0

zewelor committed:
Add device to mqtt camera (#26238)

* Add device to mqtt camera

* Support discovery device info update and add tests

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Undo accidental Tuya change

Swamp-Ig committed:
Entity registry doesn't overwrite with None (#24275)

frenck committed:
Replaces IOError by OSError (#26428)

BaQs committed:
Add atome sensor platform (#26197)

* Atome sensor platform - provides live data from Linky energy meters with Atome device from Total/Direct energie

* Proper requirements, added code ownership

* Do not cover atome component

* Proper PEP8 import, proper use of const, added missing docstring etc

* Proper PEP8 import, proper use of const, added missing docstring etc

* Integrate recommendations from MartinHjelmare

* Init shall remain as clean as possible, we don't want side effect

* Add daily,weekly,monthly,yearly sensors. Now depends on pyatome 0.1

* Requirements regenerated for atome component

* Refactored the way we update sensors

* Removed some un-necessary returns and unused variable

Commit:
[ci skip] Translation update

pvizeli committed:
Update azure-pipelines-translation.yml for Azure Pipelines

Commit:
[ci skip] Translation update

pvizeli committed:
Cleanup Dockerfile.dev (#26451)

emontnemery committed:
Add device automation condition (#26313)

* Add support for device conditions

* Lint

* Update test case

* Make and+or conditions async, adjust tests

* Cleanup tests

* Remove non callback versions of conditions, correct typing

* Correct typing

* Update light/strings.json

* Address review comments

* Make device automation lists simple lists, not dicts

* Add device_automation/const.py

* Use IS_ON/IS_OFF everywhere for conditions

exxamalte committed:
Queensland Bushfire Alert icon for geolocation entities (#26439)

* define icon

* reordered const imports

jjlawren committed:
Standardize Plex server connections (#26444)

* Common connection class

* Omit tests for new Plex files

* Oops

* Add missing properties

* Remove redundant log message

* Stopgap to avoid duplicate setups

* Cleaner check for server setup

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Cleaner check for server setup

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Not needed with previous setup check

* Remove username/password support

* Reduce log level

Co-Authored-By: Martin Hjelmare <marhje52@kth.se>

* Don't do setup in __init__

* Oops

* Committing too fast...

* Connect after init

* Catch update exceptions like media_player

* Pass in validated PlexServer instance

* Remove unnecessary check

* Counter should be unknown on init

* Remove servername config option

jjlawren committed:
Add plex server config options to media_player platform (#26458)

* Add server config options to media_player platform

* Unnecessary else

* Default host

* No need to try for default values

* Use const

emontnemery committed:
Add device automation action (#26455)

* Add support for device actions, with light as example.

* Add translation; return list

Kane610 committed:
deCONZ - use entity registry disabled_by to control available entities (#26219)

* First draft

* Support enabling disabled entities

* Clean up

* Move import

* Local entity enabled replaced during rebase

* Add option flow test

* Mark options properties with option

Commit:
[ci skip] Translation update

kellerza committed:
Add person to device_sun_light_trigger (#25877)

* Add person to device_sun_light_trigger

* tests

* fix test

balloob committed:
Bumped version to 0.98.4

balloob committed:
Merge pull request #26463 from home-assistant/rc

0.98.4

Mariusthvdb committed:
Change darksky icon for clear night (#26452)

* change icon for clear night

change mdi:weather-sunny to mdi:night for condition clear night

* changed erroneous mdi:night to mdi:weather-night

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,164 @@
+
+
+@callback
+def configured_hosts(hass):
```
We're not using this function.

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,164 @@
+        config = await box.system.get_deviceinfo()
+        box_id = config["status"]["SerialNumber"]
+
+        # Remove all other entries of hubs with same ID or host
```
Shouldn't we instead check that we don't add a box with an ID that is already configured?

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,164 @@
+            )
+
+        return self.async_create_entry(
+            title=TEMPLATE_SENSOR.format(""),
```
This will create a string with a space at the end. That's not good.

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,164 @@
+
+        errors = {}
+
+        from aiosysbus import Sysbus
```
Please move 3rd party imports to the top of the module.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,64 @@
+from collections import namedtuple
+import logging
+
+from homeassistant.components.device_tracker import DeviceScanner
```
Import and use the `ScannerEntity` instead.

https://github.com/home-assistant/home-assistant/blob/50cec91cf018bdb6a094bc1a993b80836543e96d/homeassistant/components/device_tracker/config_entry.py#L120

It has a different interface than the legacy device scanner.

Cyr-ius replied:
OK i rewrite Device Tracker module

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,164 @@
+            )
+            return await self._entry_from_box(box)
+
+        except AuthenticationRequired:
```
These exceptions are never raised.

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,210 @@
+from datetime import timedelta
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice,
```
The binary sensor entity requires a binary_sensor platform.

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,210 @@
+    def unique_id(self):
+        """Return unique_id."""
+
+        return "{}_downstream".format(self._box_id)
```
Please use f-strings instead of str.format.

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,210 @@
+            "downstream_power": self._dsl["DownstreamPower"],
+        }
+
+    @Throttle(SCAN_INTERVAL)
```
The throttle doesn't help much since we're not throttling the whole home assistant instance, just this entity instance. It's enough to define the `SCAN_INTERVAL` constant at the module level to set the default update interval.

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,210 @@
+        return "{}_downstream".format(self._box_id)
+
+    @property
+    def device_info(self):
```
Move this property to the common base class.

MartinHjelmare commented on homeassistant/components/livebox/sensor.py:
```diff
@@ -0,0 +1,210 @@
+        self._box = box
+        self._box_id = config_entry.data["box_id"]
+        self._state = None
+        self._datas = None
```
This instance attribute isn't used except as a local variable. We should remove it and just use a local variable in the update methods.

MartinHjelmare reviewed (changes requested):
What's the relationship between this integration and the Orange Livebox Play TV?

https://www.home-assistant.io/components/liveboxplaytv/

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,164 @@
+                "password": self.password,
+            },
+        )
+
```
There's no import step. Please add an import step as an entry point when importing the config yaml to the config flow.

tsvi committed:
Jewish calendar binary sensor (#26200)

* Move jewish calendar to its own platform

* Fix tests for Jewish Calendar platform

As part of this, move tests to use async_setup_component instead of
testing JewishCalendarSensor as suggested by @MartinHjelmare here:

https://github.com/home-assistant/home-assistant/pull/24958#pullrequestreview-259394226

* Get sensors to update during test

* Use hass.config.set_time_zone instead of directly calling set_default_time_zone in tests

* Cleanup log messages

* Rename result from weekly_portion to parshat_hashavua

* Fix english/hebrew tests

* Fix updating of issue melacha binary sensor

* Fix docstrings of binary sensor

* Reset timezones before and after each test

* Use correct entity_id for day of the omer tests

* Fix omer tests

* Cleanup and rearrange tests

* Remove the old issur_melacha_in_effect sensor

* Rename variables to make the code clearer

Instead of using lagging_date, use after_tzais and after_shkia

* Use dt_util.set_default_time_zone instead of hass.config.set_time_zone so as not to break other tests

* Remove should_poll set to false (accidental copy/paste)

* Remove _LOGGER messaging during init and impossible cases

* Move binary tests to standalone test functions

Move sensor tests to standalone test functions

* Collect entities before calling add_entities

* Fix pylint errors

* Simplify logic in binary sensor until a future a PR adds more sensors

* Rename test_id holyness to holiday_type

* Fix time zone for binary sensor tests

Fix time zone for sensor tests

* Don't use unnecessary alter_time in sensors

Don't use unnecessary alter time in binary sensor

Remove unused alter_time

* Simply set hass.config.time_zone instead of murking around with global values

* Use async_fire_time_changed instead of directly calling async_update_entity

* Removing debug messaging during init of integration

* Capitalize constants

* Collect all Entities before calling async_add_entities

* Revert "Don't use unnecessary alter_time in sensors"

This reverts commit 74371740eaeb6e73c1a374725b05207071648ee1.

* Use test time instead of utc_now

* Remove superfluous testing

* Fix triggering of time changed

* Fix failing tests due to side-effects

* Use dt_util.as_utc instead of reimplementing it's functionality

* Use dict[key] for default values

* Move 3rd party imports to the top of the module

* Fix imports

zxdavb committed:
bump geniushub client (#26476)

fixes #26440

SukramJ committed:
Unify device_state_attributes handling for Homematic IP Cloud (#26449)

*  unifi DSA for Homematic IP Cloud

* sabotage is not relevant for state

* TODAY_SUNSHINE_DURATION is not a group attribute

* Separated the words as requested

* add missing underscores

zewelor committed:
Add yeelight nightlight support via separate light entity (#26224)

* Add yeelight nightligh support via separate light entity

* Fix lint too many ancestors

* PR fixes

zewelor committed:
Add set scene service calls to yeelight (#26255)

* Add set scene service calls to yeelight

* Simplify code

* DRY valid brightness validation

* Fix services description

* PR fixes

Quentame committed:
Add device_info to Linky integration (#26477)

* Add device_info to Linky integration

* Remove useless SENSORS const

* Review: remove DOMAIN from the unique_id

dieselrabbit committed:
Update radiotherm climate attributes (#26465)

* Update radiotherm for lovelace

-Adds hvac_action property to better support the Lovelace Climate card.
-Clean up "fan" attribute. Now called "fan_action", as "mode" is supported via hvac_action.
-Update current operation label from "Off" to "Idle".

* black formatting

oischinger committed:
Add Viessmann ViCare Climate platform (#26151)

* Add Viessmann ViCare Climate platform

* Add water_heater and fix review comments
Update to latest PyVicare (0.1.0)

* Move PyVicare API creation to component

* More review fixes

* Return false if api creation fails

* Fix logging format

* Update PyVicare 0.1.1 to fix json issues

* Formatting fixes

flz committed:
Add new integration for Jandy iAqualink pool control (#26034)

* Import new iaqualink component with climate platform.

* Style and unittest changes, fix async_step_import.

* Reorder imports.

* Fix stale doctstrings and add unittest.

brange committed:
Add protocol and model as an optional restriction for tellstick sensors (#26026)

* Add protocl and model as restriction for tellstick sensors

A tellstick sensors is uniq identified by id, protocol and model, not
just the id.

This will add an optional restriction for "named sensors" for protocol
and model.

* Don't default config to empty string

* Compare None with 'is not'

farmio committed:
Fix KNX light tunable white rounding error (#26364)

* KNX light - tunable white rounding error

mitigate rounding errors in kelvin - mired conversion for lights with relative color temperature
fixes https://github.com/home-assistant/home-assistant/issues/26357

* typo _min_kelvin <> _max_kelvin

* black

Commit:
[ci skip] Translation update

bachya committed:
Bump aiowwlln to 2.0.1 (#26486)

oischinger committed:
Vicare: Avoid invalid temperature values (#26485)

The PyVicare API can return the string "error" in case of connection
or authentication errors.
The current_temperature value could be set to "error" instead of a
nueric value or None which breaks the climate component.

This commit sets the current_temperature to None instead.

balloob committed:
Updated frontend to 20190828.1

balloob committed:
Bumped version to 0.98.5

balloob committed:
Merge pull request #26487 from home-assistant/rc

0.98.5

scop committed:
Use PEP 526 type annotations, add some type hints (#26464)

* Add some more type hints to helpers.event

* Change most type comments to variable types

* Remove some superfluous type hints

Santobert committed:
Restructure Z-Wave Climate (#25724)

dmulcahey committed:
fix cluster configuration (#26494)

JeffLIrion committed:
Bump androidtv to 0.0.27 (#26497)

emontnemery committed:
Device automations: Rename name to entity_name in translations (#26491)

Commit:
[ci skip] Translation update

Adminiuga committed:
Bump ZHA dependencies. (#26504)

zxdavb committed:
Bump geniushubclient (#26519)

* bump client

* bump again

StevenLooman committed:
Update to async_upnp_client==0.14.11 (#26515)

cyr-ius committed:
Application of proposals

bramkragten committed:
Updated frontend to 20190908.0 (#26524)

HarlemSquirrel committed:
Add vivotek camera component (#26071)

* Add vivotek camera component

* Update vivotek camera compontent

Use async request to enable/disable motion detection

* Update Vivotek camera

- Use HTTPS
- Use IP address for still and stream

* Update vivotek component

- Add brand and model properties
- Add state property
- Use attribute to save motion detection status

* Add vivotek camera to .coveragerc

* Update vivotek camera

Fix lint errors

* Update vivotek camera

Remove unused method

* Update Vivotek integration to use libpyvivotek

Use libpyvivotek instead of directly making HTTPS API calls.

* Update Vivotek component

Address code review.

- Remove unused code
- Replace async methods with synchronous methods
- Update docstrings

* Linter fixes for Vivotek component

* Update Vivotek camera component

- Add SSL option
- Remove authentication options as only basic authentication
  is currently working

* Update Vivotek camera component

- Make frame rate configurable
- Require username and password

* Remove unused constants in Vivotek component

* Update Vivotek camera integration

- Use libpyvivotek v0.2.1 with better response parsing
- Use add_entities instead of async_add_entities

* Update Vivotek camera component

- Build camera and stream source ouside VivotekCam
- Remove unnecessary _stream_source attribute

* Update Vivotek camera component

- Move brand to constant
- Move _supported_features to property

* Update Vivotek camera compontent to remove unused property

cereal2nd committed:
Enable SolarEdge config entries (#26282)

* Initial commit for the solaredge configflow

* rerun the hassfest script

* Adding testcases

* Rerun hassfest, problem with black?

* Requirements for the tests

* Remove CONF_MONITORED_CONDITIONS from configuration.yaml

* Remove the options flow strings

* Resolve some comments

* Comments

* More comments

* Move the config from the sensor platform to the component itself

* More comments

* More comments

* Added solaredge __init__

* Added more test to increase coverage

dmulcahey committed:
use newly added is_server for cluster reporting (#26533)

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

emontnemery committed:
Add device automation support to switch entities (#26466)

* Add device automation support to switch entities

* Update switch translations

* Address review comments

emontnemery committed:
Remove device from device action schema (#26536)

flz committed:
Move iaqualink update from climate to component (#26505)

* Move iaqualink update from climate to component

* Typing fix

balloob committed:
Do not complain about automatic generated files (#26540)

* Do not complain about automatic generated files

* Update generated files

tyjtyj committed:
Fix Tuya Light without brightness (#26534)

Return none if there is brightness data from tuya

https://github.com/home-assistant/home-assistant/issues/25896

SukramJ committed:
Add Delta Counter of HmIP-SPDR to Homematic IP Cloud (#26538)

flz committed:
Add light platform to iaqualink integration (#26484)

* Add light platform to iaqualink component.

* Style changes.

* Polling moved to timer in the component

jjlawren committed:
Move config and connections to Plex component (#26488)

* Move config and connections to component

* Separate imports

* Set a unique_id on sensor

* Set a platforms const

* Add SERVERS dict, hardcode to single server

* Move to debug

* Return false

* More debug

* Import at top to fix lint

* Guard against legacy setup attempts

* Refactor to add setup callback

* Review comments

* Log levels

* Return result of callback

* Store CONFIGURING in hass.data

* Set up discovery if no config data

* Use schema to set defaults

* Remove media_player options to remove entities

* Improve error handling

tsvi committed:
Fix radiotherm local time (#26526)

We want to use our own dt_util.now() which takes into consideration the globally set DEFAULT_TIME_ZONE

oischinger committed:
Add more attributes to vicare climate entity (#26521)

Add some device information as attributes to the climate device:
 - room_temperature
 - supply_temperature
 - outside_temperature
 - active_vicare_program
 - active_vicare_mode
 - heating_curve_slope
 - heating_curve_shift
 - month_since_last_service
 - date_last_service
 - error_history
 - active_error
 - circulationpump_active

indykoning committed:
Add growatt server integration (#25635)

* Added growatt server integration

* Ran black formatter

* Processed feedback.

* Made attributes more readable.

* Create a sensor for each property.

* Added unique_ids

* Accidentally flipped functions

* Added dynamic device classes.

* Fixed stale session.

lifeisafractal committed:
Add zwave application version (#26205)

* Set application version in zwave

* update tests for more coverage

Commit:
[ci skip] Translation update

flz committed:
Remove unnecessary force_refresh=True, clarify system behavior (#26543)

* Remove unnecessary force_refresh=True, clarify system behavior

* Fix lint.

Quentame committed:
Add Météo France icons + device_class (#26441)

* Add Météo France icons

* Add Météo France device_class (temperature) + use constants

* Fix weather alert info log

* Use new f"{...} {...}" format for sensor name

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

zxdavb committed:
Bump geniushub-client to 0.6.13 (#26554)

* bump geniushub client

cyr-ius committed:
Add binary sensor & add option in config flow

MartinHjelmare commented on homeassistant/components/livebox/binary_sensor.py:
```diff
@@ -0,0 +1,79 @@
+        self._box = box
+        self._box_id = config_entry.data["box_id"]
+        self._state = None
+        self._datas = None
```
This doesn't have to be an instance attribute. We just use it as a local variable.

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,178 @@
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_ALLOW_TRACKER,
```
We should probably remove this option. The user can disable entities via the entity registry.

https://developers.home-assistant.io/docs/en/entity_registry_disabled_by.html

Why would we want to disable the whole platform as a user?

Cyr-ius replied:
Ok i remove option

cyr-ius committed:
Convert str.format to f-strings

cyr-ius committed:
fix strings for step user

cyr-ius committed:
Convert to f-string

cyr-ius committed:
Fix variable Box_ID

flz committed:
Add sensor platform to iaqualink component (#26544)

* Add sensor platform to iaqualink component

* Remove unnecessary icon, fix case where value is 0

balloob committed:
Refactor Cast (#26550)

* Refactor Cast

* Fix tests & address comments

* Update reqs

balloob committed:
Fix tests

dmulcahey committed:
fix events for smartthings acceleration cluster (#26557)

dmulcahey committed:
osram cluster (#26555)

pvizeli committed:
Nuki less strict (#26542)

Kane610 committed:
deCONZ device automations (#26366)

* Early draft

* Getting there

* Working fully with Hue dimmer remote

* Fix Balloobs comments

* No side effects in constructor

* Improve hue dimmer

* Add Ikea remote control

* Add xiaomi button support

* Refactor getting deconz event

* Added xiaomi devices and tradfri wireless dimmer

* Resolve unique id from device id

* Add Hue Tap and Tradfri on off switch

* More triggers for ikea on off switch and Aqara double wall switch

* Add support for Tradfri open close remote

* Fix changes after rebase

* Initial test

* Change id to event_id

* Fix translations and add subtypes

* Try if tests pass without the new tests

* Revert disabling tests
Add new exception InvalidDeviceAutomationConfig

* Ignore places calling remotes

* Enable all gateway tests

* Found the issue, now to identify which test creates it

* Remove block till done

* See if device automation test passes in azure

* Register event to device registry

* Enable test sensors

* Skip deconz event tests currently failing

* Added reason why skipping tests

homeassistant committed:
[ci skip] Translation update

pvizeli committed:
Ignore test output

pvizeli committed:
Improve startup of devcontainer (#26572)

pvizeli committed:
Cleanup stale script stuff (#26573)

pvizeli committed:
Bump UPC connect / fix auth issue (#26570)

* Bump UPC connect / fix auth issue

* Fix lint

* Fix platform schema

* Fix config value

* Address comment / add session cleanup

* Fix device handling

dmulcahey committed:
bump dependencies (#26576)

bramkragten committed:
Updated frontend to 20190911.0 (#26578)

tsvi committed:
Make uk_transport sensor timezone/DST aware (#26577)

* Make uk_transport sensor timezone/DST aware

* Fix offset-naive and offset-aware datetime comparison

tsvi committed:
Fix GTFS sensor wrong timezone (#26580)

flz committed:
Add switch platform to iaqualink integration (#26545)

* Add switch platform to iaqualink component

* Remove unnecessary call to constructor

dshokouhi committed:
Add Obihai integration (#26537)

* Add Obihai integration

* Lint

* Lint and bump library for multiple ports fix

* Review comments

* Review comments

* Correct errors

* Review comments

* Review comments

jjlawren committed:
Remove default host for Plex config (#26583)

* Remove default host, allow config with token(+server)

* Require one of host or token

* Oops

* Adjust schema

* Fix schema

* Add self as codeowner

* Update CODEOWNERS

balloob committed:
Home Assistant Cast (#26566)

* Add backend support for Home Assistant Cast

* Update test reqs

pvizeli committed:
Deprecate Python 3.6.0 (#26575)

* Deprecate Python 3.6.1

* Update msg

balloob committed:
Updated frontend to 20190911.1

balloob committed:
Check if git is dirty before committing (#26588)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Kane610 committed:
Remove support of UniFi device tracker configuration import (#26587)

snowzach committed:
Add support for DOODS Image Processing (#26208)

* Add support for doods

* Move connection to external module

* Fix for CI

* Another update for CI

* Reformatted via black

* Updated linting stuff

* Updated per code review

* Removed none check for something with a default

* Updated config parsing

* Updated if statements, need to disable lint check

* Fixed formatting and bug that should make linter happy

* Fixed one more issue with box drawing for areas

* removed extra imports

* Reworked per suggestion

* Changed output to debug for informational detection message

balloob committed:
Update PyChromecast (#26594)

doudz committed:
Bump zigate to 0.3.0 (#26586)

* Bump zigate to 0.3.0

homeassistant committed:
[ci skip] Translation update

BKPepe committed:
Upgrade youtube_dl to 2019.09.12.1 (#26593)

pvizeli committed:
Update azure-pipelines-release.yml

gerard33 committed:
Fix CCM messages (#26589)

pvizeli committed:
Update azure-pipelines-wheels.yml

doudz committed:
Bump zigpy-zigate to 0.3.1 (#26600)

* Bump zigpy-zigate to 0.3.1

Bump zigpy-zigate to 0.3.1 (fix Rpi.GPIO dependency)

* Bump zigpy-zigate to 0.3.1

Bump zigpy-zigate to 0.3.1 (fix Rpi.GPIO dependency)

pgilad committed:
Improve bluetooth tracker device code (#26067)

* Improve bluetooth device tracker code

* Don't use set operations

* Fix logging template interpolation

* Warn if not tracking new devices and not devices to track

* Updates due to CR

* Fix pylint warning

* Fix pylint import warning

* Merge with dev

poofyteddy committed:
Disable Watson TTS Telemetry (#26253)

homeassistant committed:
[ci skip] Translation update

SNoof85 committed:
Fix Typo (#26612)

pgilad committed:
Refactor Bluetooth Tracker to async (#26614)

* Convert bluetooth device tracker to async

* WIP

* WIP

* Fix callback

* Fix tracked devices

* Perform synchornized updates

* Add doc

* Run in executor

* Improve execution

* Improve execution

* Don't create a redundant task

* Optimize see_device to run concurrently

* Remove redundant initialization scan

pvizeli committed:
Update azure-pipelines-wheels.yml

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

Kane610 committed:
deCONZ -  create deconz_events through sensor platform (#26592)

* Move event creation into sensor platform where it belongs
* Fixed the weird failing test observed during device automation PR

homeassistant committed:
[ci skip] Translation update

amigan committed:
zha ZCL color loop effect (#26549)

* Initial implementation of ZCL color loop effect
* Fix linter complaints
* Use const for action
* Reformat with Black
* Cleanup after review.
* Handle effect being None

flz committed:
Add iaqualink binary sensor and unique_id (#26616)

* Add binary_platform to iaqualink integration, add unique_id

* Revert Mixin changes, move self.dev to AqualinkEntity

* Style fixes

SukramJ committed:
Add group attribute to Homematic IP Cloud (#26618)

* Add group attribute to Homematic IP Cloud

* Fix docstring

cyr-ius committed:
rewrite device trackers entity

cyr-ius committed:
rewrite device trackers entity

cyr-ius committed:
Remove legacy setup

Kane610 committed:
Move deCONZ services to their own file (#26645)

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,103 @@
+        try:
+            id = (await ld.async_infos())["SerialNumber"]
+
+        except Exception:
```
We should catch specific exceptions for specific problems.

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,103 @@
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
```
We can remove init if we don't use it.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Zigbee Home Automation device tracker from config entry."""
```
Stale docstring.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+        await _async_setup_entities(hass, config_entry, async_add_entities, [device])
+
+    unsub = async_dispatcher_connect(
+        hass, LIVEBOX_DISCOVERY_NEW.format(DOMAIN), async_discover
```
When do we send the signal to discover new entities?

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+async def _async_setup_entities(
+    hass, config_entry, async_add_entities, device_trackers
+):
+    """Set up the ZHA device trackers."""
```
Stale docstring.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+        self._box_id = id
+        self._ld = ld
+        self._connected = False
+        self._should_poll = True
```
Should poll is True by default. We can remove this.

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect entity object when removed."""
+
+        for unsub in self._unsubs[:]:
```
```suggestion
        for unsub in self._unsubs:
```

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+        else:
+            self._connected = False
+
+    async def _update_entity(self, update_device, device):
```
```suggestion
    async def _update_entity(self):
```

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+    async def async_update(self):
+        """Handle polling."""
+
+        if await self._update_entity(self._ld, self._device):
```
```suggestion
        if await self._update_entity():
```

MartinHjelmare commented on homeassistant/components/livebox/device_tracker.py:
```diff
@@ -0,0 +1,121 @@
+
+    async def _update_entity(self, update_device, device):
+
+        device = await self._ld.async_devices(device)
```
```suggestion
        device = await self._ld.async_devices(self._device)
```

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,149 @@
+    async def async_conn(self):
+        """Connect at the livebox router."""
+
+        box = Sysbus()
```
Do we need to create a new `Sysbus` instance each time we connect?

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,149 @@
+async def async_setup(hass, config):
+    """Load configuration for Livebox component."""
+
+    return True
```
Please add an import step to the config flow and start the config flow import step from here.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,149 @@
+                }
+            }
+        box = await self.async_conn()
+        devices = (await box.system.get_devices(parameters))["status"]["wifi"]
```
This will error if box doesn't return a box instance, eg on AuthorizationError error.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,149 @@
+
+        except AuthorizationError:
+            _LOGGER.error("User or password incorrect")
+            return False
```
This is an inconsistent return type. We should return None in this case.
```suggestion
            return None
```

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,149 @@
+            return False
+        except HttpRequestError:
+            _LOGGER.error("Http Request error to Livebox")
+            return False
```
```suggestion
            return None
```

MartinHjelmare commented:
Please solve the merge conflict.

SukramJ committed:
Add built in weather to Homematic IP Cloud (#26642)

Kane610 committed:
deCONZ - battery sensor instead of battery attribute (#26591)

* Allow all sensors to create battery sensors
* Neither binary sensor, climate nor sensor will have battery attributes

Kane610 committed:
deCONZ - Remove mechanisms to import a configuration from configuration.yaml (#26648)

chriscla committed:
Refactor nzbget to support future platform changes (#26462)

* Re-factor nzbget platform to enable future features.

* Re-factor nzbget platform to enable future features.

* Re-factor nzbget platform to enable future features.

* Re-factor nzbget platform to enable future features.

* Use pynzbgetapi instead of raw HTTP requests

* Using pynzbgetapi

* Pinning pynzbgetapi version.

* Requiring pynzbgetapi 0.2.0

* Addressing review comments

* Refreshing requirements (adding pynzbgetapi)

* Remove period from logging message

* Updating requirements file

* Add nzbget init to .coveragerc

* Adding nzbget codeowner

* Updating codeowners file

emontnemery committed:
Rename MockToggleDevice to MockToggleEntity (#26644)

* Rename MockToggleDevice to MockToggleEntity

* Fix tests

michaeldavie committed:
Fix Environment Canada weather forecast, retain icon_code sensor (#26646)

* Bump env_canada to 0.0.25

* Keep icon_code

bryanyork committed:
Emulate color temperature for non-ct lights in light groups (#23495)

* Emulate color temperature for non-ct lights in light groups

* fix tests

* Address review comments

* Fix black formatting

* Fix for pylint

* Address comments

* Fix black formatting

* Address comments

chriscla committed:
Use pynzbgetapi exceptions consistently (#26667)

Kane610 committed:
deCONZ - Improve service tests (#26663)

* Improve configure service tests

* Add refresh device service test

* Add tests for setup and unload services

* Remove refresh device test from test_init

* Extra verification of deconz services existance in hass.data

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

abmantis committed:
zha: fix 0 second transitions being ignored. (#26654)

Allow turning a light on instantly, with no transition time.
This is actually required for IKEA lights to be able to set brightness
and color temp in a single call.

cyr-ius committed:
Fix on demand

cyr-ius committed:
Fix on demand

cyr-ius committed:
Update config_flow generated

cyr-ius committed:
Update config_flow generated

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -54,25 +71,22 @@
-    await hass.config_entries.async_forward_entry_unload(config_entry, "device_tracker")
+    for component in COMPONENTS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_unload(config_entry, component)
```
Use a pattern like this:
```py
tasks = []
for component in COMPONENTS:
    tasks.append(
        hass.config_entries.async_forward_entry_unload(config_entry, component))

return all(await asyncio.gather(*tasks))
```

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,166 @@
+            }
+        box = await self.async_conn()
+        devices = await box.system.get_devices(parameters)
+        if devices.get("status", {}).get("wifi", {}):
```
```py
return devices.get("status", {}).get("wifi")
```

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,166 @@
+
+        box = await self.async_conn()
+        infos = await box.system.get_deviceinfo()
+        if infos.get("status", {}):
```
As above.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,166 @@
+
+        box = await self.async_conn()
+        status = await box.system.get_WANStatus()
+        if status.get("data", {}):
```
As above.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,166 @@
+        box = await self.async_conn()
+        parameters = {"parameters": {"mibs": "dsl", "flag": "", "traverse": "down"}}
+        dsl_status = await box.connection.get_data_MIBS(parameters)
+        if dsl_status.get("status", {}).get("dsl", {}).get("dsl0", {}):
```
As above.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,166 @@
+                password=self._entry.data["password"],
+            )
+            if await self._box.get_permissions():  # check connection successful
+                return self._box
```
Why return the box? It's an instance attribute so we have access to it in all instance methods.

MartinHjelmare commented on homeassistant/components/livebox/config_flow.py:
```diff
@@ -0,0 +1,128 @@
+import logging
+import asyncio
+from collections import namedtuple
+from aiosysbus.exceptions import AuthorizationError
```
Please add a blank line between standard library and 3rd party import groups.

MartinHjelmare commented on homeassistant/components/livebox/__init__.py:
```diff
@@ -0,0 +1,166 @@
+                    }
+                }
+            }
+        box = await self.async_conn()
```
We have to catch the potential exceptions here.

MartinHjelmare commented on homeassistant/generated/config_flows.py:
```diff
@@ -1,70 +0,0 @@
-"""Automatically generated by hassfest.
```
Please don't remove this file.

MartinHjelmare reviewed (commented):
Please don't resolve comments until the fix is applied. It's hard for the reviewer to keep track of things if a comment is resolved before the fix is pushed.

cyr-ius committed:
Add config_flow and fix return box data

homeassistant commented:
Hello @Cyr-ius,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`3b1a4a52e9f87dc7a1daf47463c8c1fcab4d1474`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;26318&#x2F;commits&#x2F;3b1a4a52e9f87dc7a1daf47463c8c1fcab4d1474) - This commit has something that looks like an email address (helo@home-assistant.io). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

Cyr-ius commented:
Sorry, I completely messed up.
I'm pull a new request
